### DISCUSSION TITLE
Stacker News Sheriff's Badge

### DISCUSSION THREAD

**Discussion by ghost** [stackernews/stacker.news#1871] [Ideas]
**Is your feature request related to a problem? Please describe.**

Stacker News introduced a cowboy hat feature a while ago, and it has seen some good adoption to date. However, the cowboy hat can be improved and augmented with a new feature called the Sheriff's Badge.

**Describe the solution you'd like**

Each day, one stacker will earn a Sheriff's Badge based on the top spender numbers from the previous day. That stacker will become the new "Stacker News Sheriff" for the next 24 hours. The username to all their posts and comments will now display a badge instead of a cowboy hat, and hovering over the badge will have the same effect as hovering over the cowboy hat (viewing the length of one's streak).

This feature will also be for status purposes (just like the cowboy hat) to start, but could evolve along with the cowboy hat to grant the stacker special privileges like extra treasure hunt opportunities each day.

When visiting a stacker's profile page, there will now be up to two icons (a cowboy hat and a sheriff's badge) with a number next to each icon. The number next to each icon should denote the total number of days that a stacker has either earned a cowboy hat or a sheriff's badge (regardless of how long their streaks were, to help those who get burned by losing long cowboy hat streaks). Competing for Sheriff's Badge streaks also doesn't make for a fun environment, as all users are now forced to compete against each other for a single Sheriff's Badge in a zero-sum match every day. The odds of any user achieving a 10+ day streak are vanishingly small.

To really make this feature fun and engaging for Stackers, Stacker News could replace the username field in the navbar with the profile picture of each stacker, and overlay a cowboy hat (near the top) or a sheriff's badge (near the lower-left corner) ontop of their photo when they earn either one of those items. The cowboy hat can even have a tilted position to notify a stacker they are at risk of losing their cowboy hat.

The same icon overlays could be applied to the profile page where stackers already see their profile photo.

**Notification Examples**

Earning a Sheriff’s Badge:

- When a fire engulfed the town’s telegraph office last night, you stepped up and saved 7 women and children from the blaze. For your courage, you’ve earned the sheriff’s badge. Nice job, partner!
- When a gang of outlaws threatened the peace of our humble town, you swiftly rode in, guns blazing, and brought them to justice. For your relentless pursuit of law and order, you've earned the sheriff's badge. That’s some fine shootin’!
- When dozens of horses were let loose on the dusty plains yesterday, you corralled them one-by-one, restoring order to the town. For your unwavering determination, you’ve earned the sheriff’s badge. Mighty fine wranglin’, hombre!
- When yesterday's heatwave caused the local water well to run dry, you stepped up to dig a new one. For bringing water and life back to the community, you've earned the sheriff's badge. 
- When last night's raging river threatened to flood the town, you quickly built a dam to divert the waters and protect our homes. For your swift action and dedication, you've earned the sheriff's badge. Fine work, trailblazer!
- When a rattlesnake infestation plagued the local merchants this week, you rounded up the venomous creatures and sent them on their way. For your daring antics, you've earned the sheriff's badge. Well done, amigo!

Losing a Sheriff’s Badge:

- You had too many drinks at the saloon last night and fired your pistol at the blackjack table. For your careless conduct, the town has revoked your sheriff’s badge until further notice. Better sharpen up, bucko.
- A riot broke out on the far side of town last night, and the raging band of looters overpowered you and your men. You’ve been forced to hand in your sheriff’s badge, looks like your luck just ran dry.
- In a moment of weakness, you were caught red-handed stealing funds from the town treasury. Town council, appalled by your betrayal, has stripped you of your sheriff's badge. Better redeem yourself, partner.
- Amidst a daring bank heist, you were outwitted by a clever gang of robbers who made off with the town's wealth. The town has asked for your sheriff's badge until the stolen funds are recovered. 
- A lightning strike set the town hall ablaze last night, and you slept through the whole thing. For your inaction and lack of awareness, the town is demanding you hand in your sheriff’s badge. Tough luck, amigo.
- You forgot to tie up the horses last night before leaving your post, and now you’re unable to defend the town and maintain order. Time to hand in your sheriff’s badge, compadre.

If the previous badge owner doesn’t zap anything on their last day of being sheriff: 

- The old sheriff @insertusername skipped town on the midnight train causing chaos far and wide. The town has nominated you to turn things around. Get to work, partner, you’ve just earned the sheriff’s badge.

**Comment by huumn** [comment]
Would you mind writing the notification text for gaining and losing a sheriff badge like for the cowboy hat?

**Comment by ghost** [comment]
> Would you mind writing the notification text for gaining and losing a sheriff badge like for the cowboy hat?

Sure, updated the original comment with notification copy
